tests/Unit/PointwiseFunctions/Xcts/Test_AdmLinearMomentum.cpp

```diff
@@ -0,0 +1,233 @@
+                          Spectral::Quadrature::GaussLobatto};
+  const Mesh<3> mesh{polynomial_order + 1, Spectral::Basis::Legendre,
+                     Spectral::Quadrature::GaussLobatto};
+
+  // Initialize surface integral
+  tnsr::I<double, 3> surface_integral;
+  for (int I = 0; I < 3; I++) {
+    surface_integral.get(I) = 0.;
+  }
+
+  // Compute integrals by summing over each element
+  for (const auto& element_id : element_ids) {
+    const auto& current_block = blocks.at(element_id.block_id());
+    const auto current_element = domain::Initialization::create_initial_element(
+        element_id, current_block, initial_ref_levels);
+
+    // Get coordinates and map
+    const auto logical_coords = logical_coordinates(mesh);
+    const ElementMap<3, Frame::Inertial> logical_to_inertial_map(
+        element_id, current_block.stationary_map().get_clone());
+    const tnsr::I<DataVector, 3> inertial_coords =
+        logical_to_inertial_map(logical_coords);
+    const auto inv_jacobian =
+        logical_to_inertial_map.inv_jacobian(logical_coords);
+    const auto det_inv_jacobian = determinant(inv_jacobian);
+
+    // Get required fields
+    const auto background_fields = solution.variables(
+        inertial_coords, mesh, inv_jacobian,
+        tmpl::list<
+            Xcts::Tags::ConformalFactor<DataVector>,
+            gr::Tags::SpatialMetric<DataVector, 3, Frame::Inertial>,
+            gr::Tags::InverseSpatialMetric<DataVector, 3, Frame::Inertial>,
+            gr::Tags::ExtrinsicCurvature<DataVector, 3, Frame::Inertial>,
+            gr::Tags::TraceExtrinsicCurvature<DataVector>>{});
+    const auto& conformal_factor =
+        get<Xcts::Tags::ConformalFactor<DataVector>>(background_fields);
+    const auto& spatial_metric =
+        get<gr::Tags::SpatialMetric<DataVector, 3, Frame::Inertial>>(
+            background_fields);
+    const auto& inv_spatial_metric =
+        get<gr::Tags::InverseSpatialMetric<DataVector, 3, Frame::Inertial>>(
+            background_fields);
+    const auto& extrinsic_curvature =
+        get<gr::Tags::ExtrinsicCurvature<DataVector, 3, Frame::Inertial>>(
+            background_fields);
+    const auto& trace_extrinsic_curvature =
+        get<gr::Tags::TraceExtrinsicCurvature<DataVector>>(background_fields);
+
+    // Compute the inverse extrinsic curvature
+    tnsr::II<DataVector, 3> inv_extrinsic_curvature;
+    tenex::evaluate<ti::I, ti::J>(make_not_null(&inv_extrinsic_curvature),
+                                  inv_spatial_metric(ti::I, ti::K) *
+                                      inv_spatial_metric(ti::J, ti::L) *
+                                      extrinsic_curvature(ti::k, ti::l));
+
+    for (auto boundary_direction : current_element.external_boundaries()) {
+      const auto face_logical_coords =
+          interface_logical_coordinates(face_mesh, boundary_direction);
+      const auto direction = Direction<3>::upper_zeta();
```

Comment:
This should be `boundary_direction`. Probably best to add a `CHECK(boundary_direction == Direction<3>::upper_zeta())` and then use only one variable.